### DISCUSSION TITLE
Shuffle for Music Player

### DISCUSSION THREAD

**Discussion by yuuiko** [ppy/osu#13039] [Ideas] [answered]
**Describe the new feature:** 
Add a shuffle button for playing music shuffled.

**Proposal designs of the feature:** 
<img width="1081" alt="osushuffle" src="https://user-images.githubusercontent.com/50823728/62191257-84815900-b3b6-11e9-9528-1d1b77855f3e.png">

**Comment by peppy** [comment]
isn't it shuffled by default?

**Comment by Gabixel** [comment]
Nope, it follows the playlist order. But a shuffle button could be a great addition, it's a normal feature for a music player. (I'm not implying that lazer is a music player, but for audio controls, this seems like a pretty welcome feature, especially when you're in the main menu.)

**Comment by yuuiko** [comment]
Let me add a little tip about its behaviour @peppy.

When the person presses the shuffle button, you’ll see the currently playing song in the list go to the top, and all the other songs in the list randomise. Then lazer will play through this shuffled list top to bottom. The shuffle button turns yellow.

This allows you to still drag around the songs to set songs to play next, as well as see whats going to play next.

When the user disables shuffle, the list returns to default alphabetical order, and the yellow is removed from the icon.

**Comment by integerrr** [comment]
Not sure if this fixed intentionally or not, but I find the music player now already has shuffled play order upon startup. Begs the questions whether a "shuffle button" is still required

**Reply by Gabixel** [reply]
The "random" order you see is, I believe, just the order in which you downloaded the beatmaps (first to last). That's how it looks in my client

**Comment by Gabixel** [comment, marked as answer]
The shuffle option was added some time ago! 🎉 This can be closed. https://github.com/ppy/osu/pull/29917

![image](https://github.com/user-attachments/assets/df7babd2-e06b-4ad2-b27c-7b62eb6d4ce1)
